src/EditorFeatures/Core/Implementation/Classification/ClassificationUtilities.cs

```diff
@@ -33,7 +33,7 @@ public static void ReturnClassifiedSpanList(List<ClassifiedSpan> list)
             s_spanCache.Enqueue(list);
         }
 
-        public static void Convert(ClassificationTypeMap typeMap, ITextSnapshot snapshot, List<ClassifiedSpan> list, Action<ITagSpan<IClassificationTag>> addTag)
+        public static void Convert(ClassificationTypeMap typeMap, ITextSnapshot snapshot, IEnumerable<ClassifiedSpan> list, Action<ITagSpan<IClassificationTag>> addTag)
         {
             foreach (var classifiedSpan in list)
```

Comment:
this is undesirable.  we specifically used List to avoid allocations in this path.

Reply:
not sure what to say. please provide how much we save here by not using IEnumerable with foreach again list here.

Reply:
so List's Enumerator is struct. so really not sure how much saving you are talking about.

Reply:
It was substantial. Please unresolved this and revert. You are now allocating where we specifically removed allocations.

Reply:
Yes, the method doesn't make any sense to me. It's called `Convert` and takes `addTag` action. What does `addTag` has to do with the conversion? It seems the method is "abstracting" a foreach loop. I don't think that's making the code better.

Reply:
I'm fine with the method being inlined.  

> @CyrusNajmabadi provide why it is substantial rather than keep repeating it is.

Because it doesn't unnecessary allocate the intermediary IEnumerator.  You are adding allocations in a codepath we deliberately went and removed allocations from because of hte perf impact.